packages/cli/src/credentials/credentials.service.ts

```diff
@@ -69,153 +69,275 @@ export class CredentialsService {
 		private readonly credentialsFinderService: CredentialsFinderService,
 	) {}
 
+	private async addGlobalCredentials(
+		credentials: CredentialsEntity[],
+		includeData: boolean,
+	): Promise<CredentialsEntity[]> {
+		const globalCredentials =
+			await this.credentialsRepository.findAllGlobalCredentials(includeData);
+
+		// Merge and deduplicate based on credential ID
+		const credentialMap = new Map(credentials.map((c) => [c.id, c]));
```

Comment:
minor: same here, Set of IDs would be more appropriate but this of course works too.

Reply:
addressed. thanks for the feedback